src/net_processing.cpp

```diff
@@ -5845,7 +5878,32 @@ bool PeerManagerImpl::SendMessages(CNode* pto)
                         }
                         if (tx_relay->m_bloom_filter && !tx_relay->m_bloom_filter->IsRelevantAndUpdate(*txinfo.tx)) continue;
                         // Send
-                        vInv.push_back(inv);
+                        bool fanout = true;
+                        const auto wtxid = txinfo.tx->GetWitnessHash();
+                        if (reconciles_txs) {
+                            auto txiter = m_mempool.GetIter(txinfo.tx->GetHash());
+                            if (txiter) {
+                                // If a transaction has in-mempool children, always fanout it.
+                                // Until package relay is implemented, this is needed to avoid
+                                // breaking parent+child relay expectations in some cases.
+                                //
+                                // Potentially reconciling parent+child would mean that for every
+                                // child we need to check if any of the parents is currently
+                                // reconciled so that the child isn't fanouted ahead. But then
+                                // it gets tricky when reconciliation sets are full: a) the child
+                                // can't just be added; b) removing parents from reconciliation
+                                // sets for this one child is not good either.
+                                if ((*txiter)->GetCountWithDescendants() <= 1) {
+                                    fanout = m_txreconciliation->ShouldFanoutTo(wtxid, pto->GetId(),
```

Comment:
can add a `LogPrint(BCLog::NET, “Non-signaling reconciliation inbound peers flooding %d Outbound peers flooding %d for debug”);` for debug purpose and observation